File: app/src/main/java/rocks/poopjournal/todont/Alarm.java
package rocks.poopjournal.todont;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class Alarm {
    private int taskId;
    private long alarmTime;
    private String frequency;

    // Constructor
    public Alarm(int taskId, long alarmTime, String frequency) {
        this.taskId = taskId;
        this.alarmTime = alarmTime;
        this.frequency = frequency;
    }

    // Getters and setters
    public int getTaskId() {
        return taskId;
    }

    public void setTaskId(int taskId) {
        this.taskId = taskId;
    }

    public long getAlarmTime() {
        return alarmTime;
    }

    public void setAlarmTime(long alarmTime) {
        this.alarmTime = alarmTime;
    }

    public String getFrequency() {
        return frequency;
    }

    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }

    // Optional: Convert alarm time to readable format (for display purposes)
    public String getFormattedAlarmTime() {
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm", Locale.getDefault());
        return sdf.format(new Date(alarmTime));
    }
}

File: app/src/main/java/rocks/poopjournal/todont/model/Alarm.kt
package rocks.poopjournal.todont.model

import rocks.poopjournal.todont.utils.Constants
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class Alarm // Constructor
    (// Getters and setters
    var habitId:Int?,var alarmTime: Long, var frequency: String
) {
    val formattedAlarmTime: String
        // Optional: Convert alarm time to readable format (for display purposes)
        get() {
            val sdf =
                SimpleDateFormat(Constants.DATE_TIME_FORMAT, Locale.getDefault())
            return sdf.format(Date(alarmTime))
        }
}

